MetasysServices/BasicServiceProvider.cs

```diff
@@ -558,8 +558,10 @@ protected async Task<List<JToken>> GetAllAvailablePagesAsync(string resource, Di
         {
             bool hasNext = true;
             List<JToken> aggregatedResponse = new List<JToken>();
-            
+
             int page = 1;
+            int pageSize = 1000;

```

Comment:
@Federico-Artioli you may want to consider exposing `pageSize` as a parameter with a default value of 100 (or 1000) in the `GetAllAvailablePagesAsync` method.

Reply:
Yes, `GetAllAvailablePagesAsync` method is already having a parameter that is of type dictionary and the purpose of the dictionary is to expose properties for example `page`, `pagesize`, etc.

In my opinion, the public methods that the MetasysServices library is exposing for example [Get all equipments](https://github.com/metasys-server/basic-services-dotnet/blob/a03b5afa5c8e4eac26569a59b624cac06fe849aa/MetasysServices/Equipments/EquipmentServiceProvider.cs#L51-L54), etc. needs to expose such properties, so that the consumer can get the data based on their provided `pagesize` values.

Reply:
Yes I also agree that exposing a parameter as `pageSize` in some public methods as `Get All Equipment` will help the user to set the required value when they call those methods.